xarray/core/formatting.py

```diff
@@ -1048,39 +1064,93 @@ def diff_datatree_repr(a: DataTree, b: DataTree, compat):
     return "\n".join(summary)
 
 
-def _single_node_repr(node: DataTree) -> str:
-    """Information about this node, not including its relationships to other nodes."""
-    if node.has_data or node.has_attrs:
-        # TODO: change this to inherited=False, in order to clarify what is
-        # inherited? https://github.com/pydata/xarray/issues/9463
-        node_view = node._to_dataset_view(rebuild_dims=False, inherited=True)
-        ds_info = "\n" + repr(node_view)
-    else:
-        ds_info = ""
-    return f"Group: {node.path}{ds_info}"
+def _inherited_vars(mapping: ChainMap) -> dict:
+    return {k: v for k, v in mapping.parents.items() if k not in mapping.maps[0]}
+
+
+def _datatree_node_repr(node: DataTree, show_inherited: bool) -> str:
+    summary = [f"Group: {node.path}"]
+
+    col_width = _calculate_col_width(node.variables)
+    max_rows = OPTIONS["display_max_rows"]
+
+    inherited_coords = _inherited_vars(node._coord_variables)
 
+    # Only show dimensions if also showing a variable or coordinates section.
+    show_dims = (
+        node._node_coord_variables
+        or (show_inherited and inherited_coords)
+        or node._data_variables
+    )
+
+    if show_dims:
+        # Includes inherited dimensions.
+        dims_start = pretty_print("Dimensions:", col_width)
+        dims_values = dim_summary_limited(
+            node, col_width=col_width + 1, max_rows=max_rows
```

Comment:
This should probably be updated to only show dimensions defined on the node (unless `show_inherited=True`)

Reply:
done